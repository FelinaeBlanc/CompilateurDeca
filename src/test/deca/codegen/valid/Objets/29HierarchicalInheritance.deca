
class A {
	void print_A() { println("Class A"); }
}

class B extends A {
	void print_B() { println("Class B"); }
}

class C extends A {
	void print_C() { println("Class C"); }
}

class D extends A {
	void print_D() { println("Class D"); }
}


{
    C obj_C = new C();
    B obj_B = new B();
    D obj_D = new D();

    obj_B.print_A();
    obj_B.print_B();
    
    obj_C.print_A();
    obj_C.print_C();
    
    obj_D.print_A();
    obj_D.print_D();
}

