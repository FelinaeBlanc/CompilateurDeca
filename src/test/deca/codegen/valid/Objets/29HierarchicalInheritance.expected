Class A
Class B
Class A
Class C
Class A
Class D
